src/main/webapp/app/shared/statistics-graph/statistics-average-score-graph.component.ts

```diff
@@ -24,121 +27,31 @@ export class StatisticsAverageScoreGraphComponent implements OnInit {
-                borderColor: GraphColors.BLUE,
-                hoverBackgroundColor: GraphColors.BLUE,
-                hoverBorderColor: GraphColors.BLUE,
-            },
-            {
-                // Average exercise score bars
-                label: this.exerciseAverageLegend,
-                data: this.exerciseAverageScores.slice(this.currentPeriod, 10 + this.currentPeriod).map((exercise) => exercise.averageScore),
-                type: 'bar',
-                backgroundColor: GraphColors.DARK_BLUE,
-                borderColor: GraphColors.DARK_BLUE,
-                hoverBackgroundColor: GraphColors.DARK_BLUE,
-            },
-        ];
-    }
-
-    private createCharts() {
-        this.barChartOptions = {
-            layout: {
-                padding: {
-                    top: 20,
-                },
-            },
-            responsive: true,
-            hover: {
-                animationDuration: 0,
-            },
-            animation: {
-                duration: 1,
-                onComplete() {
-                    const chartInstance = this.chart,
-                        ctx = chartInstance.ctx;
-                    ctx.textAlign = 'center';
-                    ctx.textBaseline = 'bottom';
-
-                    this.data.datasets.forEach(function (dataset: DataSet, j: number) {
-                        const meta = chartInstance.controller.getDatasetMeta(j);
-                        meta.data.forEach(function (bar: any, index: number) {
-                            const data = dataset.data[index];
-                            ctx.fillText(String(data), bar._model.x, bar._model.y - 5);
-                        });
-                    });
-                },
-            },
-            scales: {
-                yAxes: [
-                    {
-                        ticks: {
-                            beginAtZero: true,
-                            min: 0,
-                            max: 100,
-                        },
-                    },
-                ],
-                xAxes: [
-                    {
-                        gridLines: {
-                            offsetGridLines: false,
-                        },
-                        ticks: {
-                            autoSkip: false,
-                            callback(title: string) {
-                                return title.length > 10 ? title.substr(0, 10) + '...' : title;
-                            },
-                        },
-                    },
-                ],
-            },
-        };
+        const data = this.exerciseAverageScores.slice(this.currentPeriod, 10 + this.currentPeriod).map((exercise) => exercise.averageScore);
+        this.ngxData = [];
+        data.forEach((exerciseScore, index) => {
+            this.ngxData.push({ name: this.barChartLabels[index], value: exerciseScore });
+        });
+        this.ngxData = [...this.ngxData];
```

Comment:
Couldn't this be simplified like this? 🤔 
```suggestion
        this.ngxData = this.exerciseAverageScores.slice(this.currentPeriod, 10 + this.currentPeriod)
            .map((exercise, index) => ({ name: this.barChartLabels[index], value: exercise.averageScore }));
```

Reply:
This would make the code definitely shorter. I incorporated it at the places mentioned by you in 459d2ad